apps/astarte_appengine_api/lib/astarte_appengine_api_web/router.ex

```diff
@@ -19,106 +19,174 @@ defmodule Astarte.AppEngine.APIWeb.Router do
+    plug Astarte.AppEngine.APIWeb.Plug.JoinPath
+    plug Astarte.AppEngine.APIWeb.Plug.LogInterface
+    plug Astarte.AppEngine.APIWeb.Plug.LogPath
+  end
+
+  pipeline :interface_values_device_group_api do
+    plug :device_alias_api
+    plug Astarte.AppEngine.APIWeb.Plug.JoinPath
+    plug Astarte.AppEngine.APIWeb.Plug.LogInterface
+    plug Astarte.AppEngine.APIWeb.Plug.LogPath
+  end
+
+  pipeline :device_group_name_api do
+    plug Astarte.AppEngine.APIWeb.Plug.LogDeviceId
+    plug Astarte.AppEngine.APIWeb.Plug.LogGroupName
+  end
+
   pipeline :swagger do
     plug :maybe_halt_swagger
   end
 
   scope "/v1/:realm_name", Astarte.AppEngine.APIWeb do
-    pipe_through :api
+    pipe_through :auth_api
 
     get "/version", VersionController, :show
 
     scope "/stats" do
       get "/devices", StatsController, :show_devices_stats
     end
 
-    resources "/devices", DeviceStatusController,
-      only: [:index, :show, :update],
-      param: "device_id"
+    scope "/devices" do
+      pipe_through :device_api
 
-    resources "/devices-by-alias", DeviceStatusByAliasController,
-      only: [:index, :show, :update],
-      param: "device_alias"
+      resources "/", DeviceStatusController,
+        only: [:index, :show, :update],
+        param: "device_id"
 
-    resources "/devices/:device_id/interfaces", InterfaceValuesController,
-      only: [:index, :show],
-      param: "interface"
+      scope "/:device_id" do
+        pipe_through :interface_value_api
 
-    get "/devices/:device_id/interfaces/:interface/*path_tokens", InterfaceValuesController, :show
+        resources "/interfaces", InterfaceValuesController,
+          only: [:index, :show],
+          param: "interface"
 
-    put "/devices/:device_id/interfaces/:interface/*path_tokens",
-        InterfaceValuesController,
-        :update
+        get "/interfaces/:interface/*path_tokens", InterfaceValuesController, :show
 
-    post "/devices/:device_id/interfaces/:interface/*path_tokens",
-         InterfaceValuesController,
-         :update
+        put "/interfaces/:interface/*path_tokens",
+            InterfaceValuesController,
+            :update
 
-    delete "/devices/:device_id/interfaces/:interface/*path_tokens",
-           InterfaceValuesController,
-           :delete
+        post "/interfaces/:interface/*path_tokens",
+             InterfaceValuesController,
+             :update
 
-    resources "/devices-by-alias/:device_alias/interfaces",
-              InterfaceValuesByDeviceAliasController,
-              only: [:index, :show],
-              param: "interface"
+        delete "/interfaces/:interface/*path_tokens",
+               InterfaceValuesController,
+               :delete
+      end
+    end
 
-    get "/devices-by-alias/:device_alias/interfaces/:interface/*path_tokens",
-        InterfaceValuesByDeviceAliasController,
-        :show
+    scope "/groups/:group_name" do
+      pipe_through :interface_values_group_api
```

Comment:
Be aware that you are still declaring the same scopes multiple times, such as `/groups/:group_name`.
It's probably better to define each scope only a single time and group all corresponding routes under that single scope.
We can do an additional check on the scopes to see if they are correctly organized with a nested structure, without repeating the same scope multiple times.